lib/auth/permissions.go

```diff
@@ -149,97 +149,114 @@ func GetCheckerForBuiltinRole(role teleport.Role) (services.AccessChecker, error
 	case teleport.RoleAuth:
 		return services.FromSpec(
 			role.String(),
-			services.RoleSpecV2{
-				Namespaces: []string{services.Wildcard},
-				Resources: map[string][]string{
-					services.KindAuthServer: services.RW()},
+			services.RoleSpecV3{
+				Allow: services.RoleConditions{
+					Namespaces: []string{services.Wildcard},
+					Rules: map[string][]string{
```

Comment:
Thinking about this exact use case (when user add random resources to `Rules`) you should perhaps add check to make sure only limited user exposed resources are allowed in `Rules` that are user facing to prevent errors.